app.js

```diff
@@ -28,6 +29,21 @@ app.use((err, req, res, next) => {
   }
 });
 
+app.use((err, req, res, next) => {
+  if (err.code === "23503") {
+    res.status(404).send({ msg: "user not found" });
+  } else {
+    next(err);
+  }
+});
+app.use((err, req, res, next) => {
```

Comment:
You could combine these two error handlers into a single function. There's very little difference between this error handler and the one above :)